implementation/implementation.tex

```diff
@@ -1,69 +1,70 @@
 #3
 }
 
 \corechapter{Startup}{startup}{%
-  The startup module handles the process of starting an {\rust} application on the \gls{mcu}.
-  This involves supporting some basic requirements for the language and setting up some important initialization of the processor.
-  This module is a foundation, but can be used in isolation to run a program on the \gls{mcu} without any more dependencies.
+The \lib{startup} module handles the process of starting a {\rust} application on the \glsdesc{mcu}.
+This module is a foundation for the {\rg} platform, but it can be used in isolation to run a program on the \gls{mcu} without any dependencies.
+
+This chapter describes two important parts of a {\rust} program that needs to be configured correctly for the language to be functional with the {\gecko}.
+First, we describe how a {\rust} program can be configured to be an executable for the {\gecko}.
+At the end of this chapter we describe how we get hardware interrupts to trigger interrupt handlers implemented in {\rust}.
 }
 \include{implementation/booting-rust}
 
 \corechapter{Rust Embedded Library}{rel}{%
-  The \gls{rel} module defines the subset of the standard {\rust} library which is applicable for embedded applications.
-  This module builds on the foundation layed out by the startup module and is used by the Bindings and Application Layer modules of the {\rg} platform.
+The \glsdesc{rel} module defines the subset of the standard {\rust} library which is applicable for embedded applications.
+This module builds on the foundation laid out by the startup module and is used by the \lib{bindings} and \lib{Application Layer} modules of the {\rg} platform.
+
+This chapter starts with presenting the parts of \gls{rcl} that needs to be configured in order for the library to work for a new platform.
+We then move over to describe the standard {\rust} libraries that provides heap allocation and dynamically allocated structures.
+At the end of this, chapter we present a full overview over \gls{rel}.
 }
 \include{implementation/rust-embedded-library}
 
 
 \corechapter{Binding Libraries}{bindings}{%
-  The Bindings module includes the peripheral libraries provided by the \gls{mcu} vendor SiliconLabs, and the architecture designer ARM.
-  In order to make use of these libraries in {\rust} we developed binding libraries to expose the underlying {\C} implementation to the {\rust} language.
+The \lib{bindings} module includes the peripheral libraries provided by the \gls{mcu} vendor Silicon Labs, and the architecture designer ARM.
+In order to make use of these libraries in {\rust}, we developed binding libraries to expose the underlying {\C} implementation to the {\rust} language.
+
+We start this chapter with a section that maps the object-oriented paradigm to hardware peripherals, this lays a foundation for how we think about the hardware throughout this thesis.
+The last sections in the chapter describes how we implemented and tested the library bindings.
```

Comment:
`Troughout the last section in the chapter we descrbe how we structured, implemented, and tested the library bindings.`

Prøver å fortelle leseren at seksjon 2 er ganske lang og grunding.